init/minimal_eessi_env

```diff
@@ -20,4 +20,4 @@ fi
 export EESSI_CPU_FAMILY=$(uname -m)
 
 # set $EPREFIX since that is basically a standard in Gentoo Prefix
-export EPREFIX=$EESSI_PREFIX/compat/$EESSI_OS_TYPE/$EESSI_CPU_FAMILY
+export EPREFIX=$EESSI_COMPAT_LAYER_DIR
```

Comment:
We'll need to be a bit more careful here, we should only use `$EESSI_COMPAT_LAYER_DIR` if it's defined (via `run_in_compat_layer_env.sh` for example), but it's possible it's not set (this script is also used by the EESSI init script), so:

```suggestion
# if $EESSI_COMPAT_LAYER_DIR is defined (for example by run_in_compat_layer_env.sh script), we use that value
if [ ! -z ${EESSI_COMPAT_LAYER_DIR} ]; then
    export EPREFIX=$EESSI_COMPAT_LAYER_DIR
else
    export EPREFIX=$EESSI_PREFIX/compat/$EESSI_OS_TYPE/$EESSI_CPU_FAMILY
fi
```